### PR TITLE
add option to only generate a new file if the source is newer

### PR DESCRIPTION
**Pull request by mwarning**

### CODE REVIEW

File: misc/collect.py [hunk 1]
```diff
@@ -41,6 +46,21 @@ def change_prefix(images, old_prefix, new_prefix):
       exit(1)
     paths.append(path)
 
+# do not write data if output file is never then input data
+if args.write_new_data_only and os.path.exists(args.output):
+  p1 = os.path.getmtime(args.output)
```
Comment (aparcar) [inline misc/collect.py:51]: I would rather call it `json_output_path` and `profiles_input_path` or similar instead of p1 and p2

**Comment by aparcar** [comment]
While on it, I'd suggest to extend the handling of `generic` profiles: If the project is `generic`, and would thereby be overwritten by other profiles called generic, append a sanitized target to it.

`generic` in x86/64 should become `generic_x86-64` etc. That reflects the changes of https://github.com/openwrt/openwrt/pull/3082 and is thereby compatible with older releases like 19.07.x

File: misc/collect.py (hunk 1 above)
Comment (mwarning) [reply misc/collect.py:51]: p1 and p2 are timestamps...

Comment (aparcar) [reply misc/collect.py:51]: Well `json_output_timestamp` then?

**Comment by mwarning** [comment]
Doesn't this only affect the displayed target? I do not understand how this hides / overwrites entries in the collect.py script. Anyway, this should go into another issue.

File: misc/collect.py (hunk 1 above)
Comment (mwarning) [reply misc/collect.py:51]: `output_mtime` and `input_mtime`?

Comment (mwarning) [reply misc/collect.py]: @aparcar I made it a bit better, but not quite what you asked for... is it ok if I merge this?

**Comment by mwarning** [comment]
The build date is not from the last modified date of the overview.json anymore, making this change irrelevant.

**Pull request closed without merging**